### PR TITLE
Wor paginate

### PR DESCRIPTION
**Pull request by nicoopalenque**
# Sumary

* Paginate listing endpoints

# Card

[https://trello.com/c/Ubjmgt4b](url)

# Screenshot

Captura de Pantalla 2021-06-03 a la(s) 18 49 11

### CODE REVIEW

File: app/controllers/api/v1/open_library_controller.rb [hunk 1]
```diff
@@ -1,6 +1,12 @@
 module Api
   module V1
     class OpenLibraryController < ApplicationController
+      include Wor::Paginate
+      def index
+        book_info = OpenLibraryService.new('100').book_info
```
Comment (redwarewolf) [inline app/controllers/api/v1/open_library_controller.rb]: We could have this in constants, or env variables.

File: spec/services/open_library_service_spec.rb [hunk 2]
```diff
@@ -0,0 +1,29 @@
+require 'rails_helper'
+
+RSpec.describe 'OpenlibraryService', type: :service do
```
Comment (redwarewolf) [inline spec/services/open_library_service_spec.rb:3]: I'm more interested in the Controller's tests

Comment (redwarewolf) [reply spec/services/open_library_service_spec.rb:3]: Although these are fine too.

**Review by redwarewolf — CHANGES_REQUESTED** [review]

File: spec/services/open_library_service_spec.rb (hunk 2 above)
Comment (nicoopalenque) [reply spec/services/open_library_service_spec.rb:3]: These tests are focused on the open library service response, for example, this:

[ https://openlibrary.org/api/books?bibkeys=ISBN:0385472579&format=json&jscmd=data](url)

File: app/controllers/api/v1/open_library_controller.rb [hunk 3]
```diff
@@ -1,6 +1,13 @@
 module Api
   module V1
     class OpenLibraryController < ApplicationController
+      include Wor::Paginate
```
Comment (blacksam07) [inline app/controllers/api/v1/open_library_controller.rb]: move this to ApplicationController

File: app/controllers/api/v1/open_library_controller.rb [hunk 4]
```diff
@@ -1,6 +1,13 @@
...
+      def index
+        @isbn = '100'
```
Comment (blacksam07) [inline app/controllers/api/v1/open_library_controller.rb]: as a constant outside the method 
```ruby
ISBN = '100'.freeze # freeze method block the constant value 
```

File: app/models/open_library.rb [hunk 5]
```diff
@@ -0,0 +1,2 @@
+class OpenLibrary < ApplicationRecord
```
Comment (blacksam07) [inline app/models/open_library.rb]: 👀 why create this model?

File: config/routes.rb [hunk 6]
```diff
@@ -3,7 +3,9 @@
   namespace :api do
     namespace :v1 do
       mount_devise_token_auth_for 'User', at: 'auth', skip: [:omniauth_callbacks]
-      resources :open_library, only: :show, param: :isbn
+      resources :open_library, only: [:index, :show], param: :isbn
     end
   end
+
+  get 'paginate', to: 'dummy_models#index'
```
Comment (blacksam07) [inline config/routes.rb]: 👀 why expose this route?

**Comment by redwarewolf** [comment]
I have nothing to add to Samir's review, when all those issues are solved I'll approve =)

**Review by blacksam07 — APPROVED** [review]

**Pull request merged**